File: TP2-Python/TP2-Kode Program/produk.py

class Produk:
    def __init__(self, id_produk, asal_produk, harga):
        # Atribut menggunakan underscore untuk menandakan "protected"
        self._id_produk = id_produk
        self._asal_produk = asal_produk
        self._harga = harga

    # Getter methods
    def get_id_produk(self):
        return self._id_produk

    def get_asal_produk(self):
        return self._asal_produk

    def get_harga(self):
        return self._harga

File: TP2-PHP/Kode Program/Produk.php
<?php
class Produk {
    protected $id_produk;
    protected $asal_produk;
    protected $harga;

    public function __construct($id_produk, $asal_produk, $harga) {
        $this->id_produk = $id_produk;
        $this->asal_produk = $asal_produk;
        $this->harga = $harga;
    }

    //getter method
    public function getIdProduk() {
        return $this->id_produk;
    }

    public function getAsalProduk() {
        return $this->asal_produk;
    }

    public function getHarga() {
        return $this->harga;
    }

}
